Excluding getter-setter from sonar

biswassri opened the pull request.

rareddy commented on api/src/main/java/org/bf2/operator/Generated.java:
```diff
@@ -0,0 +1,15 @@
+package org.bf2.operator;
```
IMO this class should probably reside in common module in some util or code package

MikeEdgar replied:
We're already using Lombok for equals/hashcode implementations, I think I would rather use it for getters/setters than going with the annotation. The "generated" naming that is required for the annotation is a bit of a smell for code that isn't really generated.

biswassri replied:
Oh, we are using Lombok? I couldn't find a Lombok.config so I thought this would be a better approach. @MikeEdgar  where should I look to find the lombok.config?

MikeEdgar replied:
We're using it, but the config is optional unless you want to do something like used `lombok.addLombokGeneratedAnnotation = true`. You can add the config along with your change.

biswassri replied:
Yeah I was hoping to add that, so for further classes, the annotation gets added. Does that sound good?

MikeEdgar replied:
Yes, that's good. You can go ahead and remove any simple getter/setter methods in the `api` project and add the appropriate Lombok annotations to generate them as well.

MikeEdgar commented on api/src/main/java/org/bf2/operator/resources/v1alpha1/Versions.java:
```diff
@@ -29,35 +31,14 @@ public class Versions {
     public static final String STRIMZI_CLUSTER_OPERATOR_V0_23_0_4 = "strimzi-cluster-operator.v0.23.0-4";
 
     @NotNull
+    @Getter @Setter
```
What happens with these on the class instead of being on the fields?

biswassri replied:
It generates for only those fields. But I think I can move it to class level here as well since the other variables are static and final.

MikeEdgar commented on pom.xml:
```diff
@@ -328,6 +328,7 @@
 	                    <!-- exclude entry points -->
 	                    <exclude>**/KasFleetShardOperator.class</exclude>
 	                    <exclude>**/Main.class</exclude>
+                        <exclude>**/**/ExcludedPOJO.class</exclude>
```
Is this needed?

MikeEdgar commented on api/src/main/java/org/bf2/operator/resources/v1alpha1/ManagedKafkaCapacity.java:
```diff
@@ -18,14 +20,14 @@
-    private Integer maxPartitions;
-    private String maxDataRetentionPeriod;
-    private Integer maxConnectionAttemptsPerSec;
+    @Setter private Quantity ingressPerSec;
+    @Setter private Quantity egressPerSec;
+    @Getter @Setter private Quantity ingressEgressThroughputPerSec;
+    @Getter @Setter private Integer totalMaxConnections;
+    @Getter @Setter private Quantity maxDataRetentionSize;
+    @Getter @Setter private Integer maxPartitions;
+    @Getter @Setter private String maxDataRetentionPeriod;
+    @Getter @Setter private Integer maxConnectionAttemptsPerSec;
```
If these are on the class, I think Lombok should ignore the getter methods that already are implemented.

biswassri replied:
Yeah that makes sense! Let me try that.

biswassri commented:
@MikeEdgar I noticed, after I removed `ExcludedPOJO.class` from the pom the coverage went down by 8%, so I added it back. Also, trying to add `<sonar.issue.ignore.allfile>@Generated</sonar.issue.ignore.allfile>` for removing the generated class. I wanted to suggest, what do you think about removing `api` module completely as well from coverage?

MikeEdgar commented:
> after I removed `ExcludedPOJO.class` from the pom the coverage went down by 8%
There's something strange about that - there is no such class in the project. I suspect it is something else in that same commit that caused the decrease.

> what do you think about removing api module completely as well from coverage?
Once the getter/setter methods are excluded, there is very little code left and I think most of it is covered. One of the earlier pushes I think had the coverage up to 62% or so right?

biswassri commented:
> Once the getter/setter methods are excluded, there is very little code left and I think most of it is covered. One of the earlier pushes I think had the coverage up to 62% or so right?

Right! at commit `[81a3a6d](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/pull/809/commits/81a3a6dbf05765d11eb3e4ffd311a2cc4234c6a7)` it went up to 62%. Once I moved all the `getter/setter` to a class level it went back to 59%. So I'm not sure, what's causing the various.

MikeEdgar reviewed (approved):
LGTM - coverage increases from 59% to almost 66%

rareddy reviewed (approved):
lgtm
